### PR TITLE
add supervisor_id to dashboard tables

### PR DESCRIPTION
**Pull request by sravfeyn**
@snopoke

### CODE REVIEW

File: custom/icds_reports/migrations/0096_supervisor_id_to_dashboard_ccs_record.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        ),
+        migrations.AddField(
+            model_name='aggregateccsrecordthrforms',
+            name='supervisor_id',
+            field=models.CharField(max_length=40, null=True),
+        ),
+        migrations.AddField(
+            model_name='aggregatecomplementaryfeedingforms',
+            name='supervisor_id',
+            field=models.TextField(null=True),
```
Comment (snopoke) [inline custom/icds_reports/migrations/0096_supervisor_id_to_dashboard_ccs_record.py]: why a text field here and a char field elsewhere?

File: custom/icds_reports/utils/aggregation_helpers/complementary_forms.py [hunk 2]
```diff
@@ -84,7 +85,8 @@ def aggregation_query(self):
             END AS diet_quantity,
             CASE WHEN ucr.latest_time_end IS NOT NULL
                  THEN ucr.hand_wash ELSE prev_month.hand_wash
-            END AS hand_wash
+            END AS hand_wash,
+            supervisor_id
```
Comment (snopoke) [inline custom/icds_reports/utils/aggregation_helpers/complementary_forms.py]: generally prefer to put this at the top with state_id (same comment in other places)

File: custom/icds_reports/migrations/0096_supervisor_id_to_dashboard_ccs_record.py (hunk 1 above)
Comment (sravfeyn) [reply custom/icds_reports/migrations/0096_supervisor_id_to_dashboard_ccs_record.py]: I think I fixed it as you were reviewing.

File: custom/icds_reports/utils/aggregation_helpers/complementary_forms.py (hunk 2 above)
Comment (sravfeyn) [reply custom/icds_reports/utils/aggregation_helpers/complementary_forms.py]: Sure, why not. Updated.

File: custom/icds_reports/utils/aggregation_helpers/aww_incentive.py [hunk 3]
```diff
@@ -42,6 +42,7 @@ def aggregation_query(self):
             %(month)s AS month,
             awcm.awc_id,
             awcm.block_id,
+            awcm.supervisor_id,
```
Comment (snopoke) [inline custom/icds_reports/utils/aggregation_helpers/aww_incentive.py]: you'll need to add this to the group by clause

**Review by calellowitz — APPROVED** [review]
Looks good assuming these have already been added to the underlying UCRs

File: custom/icds_reports/utils/aggregation_helpers/aww_incentive.py (hunk 3 above)
Comment (sravfeyn) [reply custom/icds_reports/utils/aggregation_helpers/aww_incentive.py]: (Done)

**Review by snopoke — COMMENTED** [review]
Looks good. Please can you confirm that all the tables we're selecting supervisor_id from actually exist.

**Comment by sravfeyn** [comment]
Since we are updating the only dashboard tables where we added supervisor_id to the underlying UCR tables, these all should have it. But I think this should be merged after the data is backfilled?

Also, are there tests for these, that I can update the test data for supervisor_id and make sure I haven't broken any queries?

**Comment by snopoke** [comment]
Yes we'll have to wait till backfill is done before merging.

See comment here re tests: https://github.com/dimagi/commcare-hq/pull/23331#discussion_r259290910

File: custom/icds_reports/migrations/0100_add_supervisor_id.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+    dependencies = [
+        ('icds_reports', '0099_service_delivery_report'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='aggregatebirthpreparednesforms',
+            name='supervisor_id',
+            field=models.CharField(max_length=40, null=True),
```
Comment (snopoke) [inline custom/icds_reports/migrations/0100_add_supervisor_id.py]: these must all be `Text` to match the UCRs so that we get co-location with citusdb

**Review by snopoke — APPROVED** [review]

**Comment by sravfeyn** [comment]
Thanks for the fixes @snopoke. I am waiting to get https://github.com/dimagi/commcare-hq/pull/23439 to pass and then will update those tests in this PR again before merging.

**Comment by snopoke** [comment]
@sravfeyn I had second thoughts about the tests and pilled @emord on it as well who agreed that it's maybe not the best idea: https://github.com/dimagi/commcare-hq/pull/23439#issuecomment-467915277

Sorry to waste your time on those. I also checked the queries that we changed manually to confirm they produce the same data.

So I think we can merge this (and #23331 when tests pass)

**Pull request merged**